functions/Get-DbaUserPermission.ps1

```diff
@@ -215,17 +215,65 @@ function Get-DbaUserPermission {
                 $dbs = $dbs | Where-Object IsSystemObject -eq $false
             }
 
-            #reset $serverDT
-            $serverDT = $null
+            # grab server data (just the one time):
+            Write-Message -Level Verbose -Message "Building data table for server objects"
 
-            foreach ($db in $dbs) {
-                Write-Message -Level Verbose -Message "Processing $db on $instance"
+            $sqlFile = Join-Path -Path $script:PSModuleRoot -ChildPath "bin\stig.sql"
+            $sql = [System.IO.File]::ReadAllText("$sqlFile")
+            $sql = $sql.Replace("<TARGETDB>", $master.Name)
 
-                $db.ExecuteNonQuery($endSQL)
+            try {
+                $master.ExecuteNonQuery($sql)
+            } catch {
+                # here to avoid an empty catch
+                $null = 1
+            } # sometimes it complains about not being able to drop the stig schema if the person Ctrl-C'd before.
 
-                if ($db.IsAccessible -eq $false) {
-                    Stop-Function -Message "The database $db is not accessible" -Continue
+            try {
+                $serverDT = $master.Query($serverSQL)
+
+                foreach ($row in $serverDT) {
+                    [PSCustomObject]@{
+                        ComputerName       = $server.ComputerName
+                        InstanceName       = $server.ServiceName
+                        SqlInstance        = $server.DomainInstanceName
+                        Object             = 'SERVER'
+                        Type               = $row.Type
+                        Member             = $row.Member
+                        RoleSecurableClass = $row.'Role/Securable/Class'
+                        SchemaOwner        = $row.'Schema/Owner'
+                        Securable          = $row.Securable
+                        GranteeType        = $row.'Grantee Type'
+                        Grantee            = $row.Grantee
+                        Permission         = $row.Permission
+                        State              = $row.State
+                        Grantor            = $row.Grantor
+                        GrantorType        = $row.'Grantor Type'
+                        SourceView         = $row.'Source View'
+                    }
                 }
+            } catch {
+                Write-Message -Level warning "An error occured while retrieving server permissions."
+            }
+
+            #delete objects from running server permissions query
```

Comment:
Why do you have to do that?

Reply:
I wasn't looking to completely re-factor this function: I simply wanted to fix an issue where if run against a database that was not accessible it threw an error and failed. 

The [current version of the function](https://github.com/dataplat/dbatools/blob/development/functions/Get-DbaUserPermission.ps1) Currently acts in the same way: creating the objects, running the query and deleting the objects for each database it is running against on an instance. If there is a specific fix you would like to see here please feel free to update it or tell me specifically what you would like to see.

Reply:
You are (mostly) right here. And executing the $endSQL in the contect of tempdb would clean up after the usage of STIG. But to avoid left-overs from another run, cleanup is done at the beginning, so once at the end of the instance would be enough.